promql/engine.go

```diff
@@ -601,6 +618,12 @@ func (ng *Engine) execEvalStmt(ctx context.Context, query *query, s *EvalStmt) (
 	}
 	evalSpanTimer.Finish()
 
+	if currentSamples = evaluator.currentSamples; currentSamples > query.sampleStats.PeakSamples {
```

Comment:
I would deal with that in the stats package; here I would just do query.sampleStats,ObserveSamples(evaluator.currentSamples)

and in querystat that would set the peak, total, and increase by 1 evaluator.

Reply:
Sure. Will do this change.

Reply:
Done. Please have a look @roidelapluie.